gammapy/cube/tests/test_models.py

```diff
@@ -82,21 +82,28 @@ def diffuse_evaluator(diffuse_model, exposure, background, psf, edisp):
     return MapEvaluator(diffuse_model, exposure, background, psf=psf, edisp=edisp)
 
 
-class TestSkyModels:
-    def setup(self):
-        self.sky_models = SkyModels([sky_model(), sky_model()])
+@pytest.fixture(scope="session")
+def sky_models(sky_model):
+    return SkyModels([sky_model, sky_model])
 
-    def test_to_compound_model(self):
-        sky_models = self.sky_models
+
+@pytest.fixture(scope="session")
```

Comment:
Models parameter values can chance in analyses and models are very cheap to create.
I usually only put `scopy="session"` on fixtures for objects that should be read-only (like loaded datasets, even if Python doesn't pretect against accidental changes) and that take aren't super-cheap to create (say fixture execution takes at least a millisecond or a few).

Reply:
It might not be the case now, but if someone adds a second test that uses this model, then the results fill usually depend on test execution order and the second test that runs finds the model parameters in a non-clear state.

Reply:
Thanks for the explanation, it makes sense to me. I'll review the use of the `scope` argument now again...